### PR TITLE
MOD: Bug fix.

### PR DESCRIPTION
**Pull request by JueShanCoder**
MOD: Bug fix.

### CODE REVIEW

File: contracts/influencemining/Auction.sol [hunk 1]
```diff
@@ -23,13 +23,23 @@ contract Auction is OwnableUpgradeable {
...
+        address governanceTokenAddr;
     }
 
     struct HNFTInfo {
         uint256 hNFTId;
         address hNFTContractAddr;
     }
 
+    struct PrepareBidInfo {
+        uint256 bidId;
```
Comment (coderfengyun) [inline contracts/influencemining/Auction.sol:35]: bidId -> newBidId

File: contracts/influencemining/Auction.sol [hunk 2]
```diff
@@ -49,24 +59,26 @@ contract Auction is OwnableUpgradeable {
...
-        require(hNFTContract.getApproved(hNFTId) == address(this), "hNFTId does not approve");
+        IERC721H hNFTContract = IERC721H(hNFTContractAddr);
+        require(hNFTContract.isSlotAuthorized(hNFTId, address(this)), "not slotManager");
         IERC20 ad3Add = IERC20(ad3Address);
         require(ad3Add.balanceOf(_msgSender()) >= MIN_DEPOIST_FOR_PRE_BID, "AD3 balance not enough");
         require(ad3Add.allowance(_msgSender(), address(this)) >= MIN_DEPOIST_FOR_PRE_BID, "allowance not enough");
         ad3Add.transferFrom(_msgSender(), address(this), MIN_DEPOIST_FOR_PRE_BID);
         uint256 preBidId = _generateRandomNumber();
-        preBids[hNFTContractAddr][hNFTId]= PreBid(preBidId, MIN_DEPOIST_FOR_PRE_BID, _msgSender(), block.timestamp);
+        address governanceTokenAddr = HNFTGovernance(hnftGoverAddress).getGovernanceToken(hNFTContractAddr, hNFTId);
```
Comment (coderfengyun) [inline contracts/influencemining/Auction.sol:76]: 增加测试用例来验证governanceToken逻辑

File: contracts/influencemining/Auction.sol (hunk 1 above)
Comment (JueShanCoder) [reply contracts/influencemining/Auction.sol:35]: done

**Review by coderfengyun — APPROVED** [review]

File: contracts/influencemining/Auction.sol (hunk 2 above)
Comment (JueShanCoder) [reply contracts/influencemining/Auction.sol:76]: done.

**Pull request merged**